apps/prairielearn/elements/pl-file-upload/pl-file-upload.py

```diff
@@ -37,48 +110,84 @@ def add_format_error(data: pl.QuestionData, error_string: str) -> None:
 
 def prepare(element_html: str, data: pl.QuestionData) -> None:
     element = lxml.html.fragment_fromstring(element_html)
-    required_attribs = ["file-names"]
-    optional_attribs = []
+
+    # Either "file-names" or "optional-file-names" is required, which is checked separately
+    required_attribs = []
+    optional_attribs = ["file-names", "optional-file-names", "allow-blank"]
```

Comment:
I believe it should be an error to use "allow-blank=true" and a non-blank "file-names". Note that this will cause a parse error anyway, as "file-names" will tag the files are required. I propose that `allow-blank` should only be used with `optional-file-names`.

Reply:
Added a check.